tools/cmake/toolchains/GCC_ARM.cmake

```diff
@@ -7,70 +7,61 @@ set(CMAKE_CXX_COMPILER "arm-none-eabi-g++")
 set(GCC_ELF2BIN "arm-none-eabi-objcopy")
 set_property(GLOBAL PROPERTY ELF2BIN ${GCC_ELF2BIN})
 
-# Sets toolchain options
+# Sets toolchain options for a target
 function(mbed_set_toolchain_options target)
-    list(APPEND link_options
-        "-Wl,--start-group"
-            "-lstdc++"
-            "-lsupc++"
-            "-lm"
-            "-lc"
-            "-lgcc"
-            "-lnosys"
-        "-Wl,--end-group"
-        "-specs=nosys.specs"
+    target_link_options(${target} INTERFACE
         "-T" "${CMAKE_BINARY_DIR}/${APP_TARGET}.link_script.ld"
-        "-Wl,-Map=${CMAKE_BINARY_DIR}/${APP_TARGET}.map"
-        "-Wl,--cref"
-    )
+        "-Wl,-Map=${CMAKE_BINARY_DIR}/${APP_TARGET}.map")
+endfunction(mbed_set_toolchain_options)
 
-    # Add linking time preprocessor macro for TFM targets
-    if("TFM" IN_LIST MBED_TARGET_LABELS)
-        list(APPEND link_options
-            "-DDOMAIN_NS=1"
-        )
-    endif()
+# build toolchain flags that get passed to everything (including CMake compiler checks)
+list(APPEND link_options
+    "-Wl,--start-group"
+        "-lstdc++"
+        "-lsupc++"
+        "-lm"
+        "-lc"
+        "-lgcc"
+        "-lnosys"
+    "-Wl,--end-group"
+    "-specs=nosys.specs"
+    "-Wl,--cref")
 
-    list(APPEND common_options
-        "-Wall"
-        "-Wextra"
-        "-Wno-unused-parameter"
-        "-Wno-missing-field-initializers"
-        "-fmessage-length=0"
-        "-fno-exceptions"
-        "-ffunction-sections"
-        "-fdata-sections"
-        "-funsigned-char"
-        "-MMD"
-        "-fomit-frame-pointer"
-        "-g3"
+# Add linking time preprocessor macro for TFM targets
+if("TFM" IN_LIST MBED_TARGET_LABELS)
+    list(APPEND link_options
+        "-DDOMAIN_NS=1"
     )
+endif()
 
-    target_compile_options(${target}
-        INTERFACE
-            ${common_options}
-    )
+list(APPEND common_options
+    "-Wall"
+    "-Wextra"
+    "-Wno-unused-parameter"
+    "-Wno-missing-field-initializers"
+    "-fmessage-length=0"
+    "-fno-exceptions"
+    "-ffunction-sections"
+    "-fdata-sections"
+    "-funsigned-char"
+    "-fomit-frame-pointer"
+    "-g3"
+)
 
-    target_link_options(${target}
-        INTERFACE
-            ${common_options}
-            ${link_options}
-    )
-endfunction()
 
 # Configure the toolchain to select the selected C library
 function(mbed_set_c_lib target lib_type)
     if (${lib_type} STREQUAL "small")
         target_compile_definitions(${target}
             INTERFACE
-                MBED_RTOS_SINGLE_THREAD
-                __NEWLIB_NANO
-        )
+            MBED_RTOS_SINGLE_THREAD
+            __NEWLIB_NANO
+            )
```

Comment:
Please revert this.

Reply:
Dang, CLion has no setting for putting the closing paren at the outer indent level, and in fact it auto indents it whenever I paste code...  Do you have any way to auto reformat these?

Reply:
We do not yet have auto formatter in place, if you got any suggestions , we would like to try it out.